test/hint.lean

```diff
@@ -0,0 +1,17 @@
+import Mathlib.Tactic.Common
+import Mathlib.Tactic.Linarith
+import Mathlib.Data.Nat.Prime
+
+example (h : 1 < 0) : False := by hint
```

Comment:
Could we use `#guard_msgs` to see what these are testing?

Reply:
Sure. I'm a little worried that once I parallelize the tactic execution the output will become nondeterministic, but we can cross that bridge later.

Reply:
There's also one that will change once `simp` stops calling `decide`, but again, I'll fix that later.